src/networkd.c

```diff
@@ -809,6 +818,41 @@ write_wpa_conf(const NetplanNetDefinition* def, const char* rootdir)
         g_hash_table_iter_init(&iter, def->access_points);
         while (g_hash_table_iter_next(&iter, NULL, (gpointer) &ap)) {
             g_string_append_printf(s, "network={\n  ssid=\"%s\"\n", ap->ssid);
+            if (ap->bssid) {
+                g_string_append_printf(s, "  bssid=%s\n", ap->bssid);
+            }
+            if (ap->band) {
+                if (ap->band == NETPLAN_WIFI_BAND_24) {
+                    // initialize frequency hashtable
+                    if(!wifi_frequency_24)
+                        wifi_get_freq24(1);
+                    if (ap->channel) {
+                        g_string_append_printf(s, "  freq_list=%d\n", wifi_get_freq24(ap->channel));
+                    } else {
+                        g_string_append_printf(s, "  freq_list=");
+                        g_hash_table_foreach(wifi_frequency_24, wifi_append_freq, s);
+                        // overwrite last whitespace with newline
+                        s = g_string_overwrite(s, s->len-1, "\n");
+                    }
+                }
+                if (ap->band == NETPLAN_WIFI_BAND_5) {
```

Comment:
I suppose this would be better of becoming an 'else if'.
I was also thinking of maybe making this less boilerplate, as we basically have the same code (with different variables) executed twice, but then again it would become less readable. So maybe let's leave it as is for now.

Reply:
Right. And the surrounding "if(ap->band)" can be removed, too.